drivers/sdhc/sdhc_spi.c

```diff
@@ -631,13 +633,22 @@ static int sdhc_spi_request(const struct device *dev,
 				/* CMD12 is required after multiple read, or
 				 * to retry failed transfer
 				 */
-				sdhc_spi_send_cmd(dev,
+				stop_ret = sdhc_spi_send_cmd(dev,
 					(struct sdhc_command *)&stop_cmd,
 					false);
+				while ((stop_ret != 0) && (retries < 0)) {
+					/* Retry stop command */
+					ret = stop_ret = sdhc_spi_send_cmd(dev,
+						(struct sdhc_command *)&stop_cmd,
+						false);
+					retries--;
```

Comment:
changing the same variable retries in so many places is making this code confusing to read for me

Reply:
Is there a way you'd rather structure this? As far as I know, only two locations in this function decrement the value

Reply:
i guess it's the changing and the checking, it feels like this code is doing 2 different things with one variable, there is retries for sending the stop command and retries for the request as a whole, should these be two different counts?

Reply:
Yeah I see the concern- given how we document the `retries` field, having a separate retry counter for this makes sense- I'll add one.